src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -760,6 +726,24 @@ internal void IncrementConnectionCount()
             }
         }
 
+        private bool TransferConnection(HttpConnection connection)
+        {
+            Debug.Assert(Monitor.IsEntered(SyncObj));
+
+            while (HasWaiter())
+            {
+                ConnectionWaiter waiter = DequeueWaiter();
+                if (waiter.TransferConnection(connection))
+                {
+                    return true;
+                }
+
+                // Couldn't transfer to that waiter because it was cancelled.
```

Comment:
Maybe Debug.Assert that?  Just to help to detect an erroneous condition where it was somehow successfully completed by someone else but still left in the list.

Reply:
This is asserted in ConnectionWaiter.TransferConnection:

https://github.com/dotnet/corefx/pull/32297/files/9a131eebcf24735992ecb82bdd36900278b62a1c#diff-757ede4c7814b64fabf2b422ec97a4e9R1105